### PR TITLE
[Bug Fixed] Fix all reduce grads when training more than one models

### PR DESCRIPTION
**Pull request by wkcn**
**Description**
Fix the issue #62 

**Major Revision**
- Remove `model_state.ready_to_all_reduce_grads` check in optimizer
- unit-test

### CODE REVIEW

**Review by guoshzhao — APPROVED** [review]

File: tests/optim/test_adamw.py [hunk 1]
```diff
@@ -72,6 +72,34 @@ def check_optimizer_step(self, optimizer_class, diff=3e-4):
 
         self.assertTrue(torch.allclose(model1.weight, model2.weight.float(), 0, diff))
 
+    def test_all_reduce_grads(self):
+        """Test the function `all_reduce_grads`."""
+        from msamp.common.tensor import TensorDist
```
Comment (tocean) [inline tests/optim/test_adamw.py:77]: Shall we move the "import" statement to the front of this file?

**Comment by tocean** [comment]
Do we have the scenario of training multiple models?

**Comment by wkcn** [comment]
> Do we have the scenario of training multiple models?

https://github.com/facebookresearch/fairseq/blob/main/fairseq/trainer.py#L928-L930
```python
                self.optimizer.all_reduce_grads(self.model)
                if utils.has_parameters(self.criterion):
                    self.optimizer.all_reduce_grads(self.criterion)
```
In fairseq, the gradients of two modules `self.model` and `self.criterion` are both reduced.

**Review by tocean — APPROVED** [review]
LGTM

**Pull request merged**